Marlin/src/HAL/shared/eeprom_i2c.cpp

```diff
@@ -35,44 +35,50 @@
 #include "../HAL.h"
 #include <Wire.h>
 
+#ifndef EEPROM_WRITE_DELAY
+  #define EEPROM_WRITE_DELAY    5
+#endif
+
 // ------------------------
 // Private Variables
 // ------------------------
 
-static constexpr uint8_t eeprom_device_address = I2C_ADDRESS(0x50);
+#ifndef EEPROM_DEVICE_ADDRESS
+  #define EEPROM_DEVICE_ADDRESS  0x50
+#endif
+
+static constexpr uint8_t eeprom_device_address = I2C_ADDRESS(EEPROM_DEVICE_ADDRESS);
 
 // ------------------------
 // Public functions
 // ------------------------
 
-static void eeprom_init() {
-  Wire.begin();
-}
+static void eeprom_init() { Wire.begin(); }
 
 void eeprom_write_byte(uint8_t *pos, unsigned char value) {
-  unsigned eeprom_address = (unsigned) pos;
-
-  eeprom_init();
```

Comment:
So… What I did as part of the cleanup was to remove this `eeprom_init` just for the LPC platform since that was the overall effect of your changes. And then Chris mentioned that it was not wanted, so now I have removed it completely. If this is actually (still) needed to work around some other issue then we'll find out quickly after deployment.

Reply:
Ok, originally these inits were included in 4b16fa3272032a66200515750608f35dffbf1c09 by @p3p. I think it's safe to assume they were not included to fix something and are not needed.